onnxscript/ir/passes/common/unused_removal.py

```diff
@@ -104,18 +104,32 @@ def __init__(self, remove_initialized_inputs: bool = False):
         super().__init__()
         self.remove_initialized_inputs = remove_initialized_inputs
 
+    def _remove_unused_initialized_inputs(self, model: ir.Model) -> None:
+        graph_outputs = model.graph.outputs
+        initializers = model.graph.initializers
+        graph_inputs = model.graph.inputs
+        unused_init_inputs = []
+        for i, inp in reversed(list(enumerate(graph_inputs))):
+            if inp.name in initializers and not (inp.uses() or inp in graph_outputs):
+                if self.remove_initialized_inputs:
+                    del graph_inputs[i]
```

Comment:
Why would a user want to remove a graph-input? It is part of its signature. Even if the input is never used. It could potentially fail in a system when the user specifies an input-value, and the system rejects it as an invalid input (eg., could happen with onnxruntime). Against this disadvantage, there is nothing much gained by removing the model input.

The initializer value can, however, be removed, if there is no use of that value within the model. May be that will save some memory for large initializers.

Reply:
I think the scenario is that when an initializer happens to be declared an input (as it can be in some converters), if only the initializer is removed but not the input, the same calling will fail. A user may decide to not care about those inputs since for example in ort inputs are provided by keyword.